crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1471,9 +1471,10 @@ fn is_unicode_identifier_start(c: char) -> bool {
 // Checks if the character c is a valid continuation character as described
 // in https://docs.python.org/3/reference/lexical_analysis.html#identifiers
 fn is_identifier_continuation(c: char) -> bool {
-    match c {
-        'a'..='z' | 'A'..='Z' | '_' | '0'..='9' => true,
-        c => is_xid_continue(c),
+    if c.is_ascii() {
```

Comment:
The fast path here is definitely subtle. I might add a short comment explaining the `is_ascii()` check here. I think it would be pretty easy for someone to come along and remove the check because it is, logically speaking, redundant. Maybe something like:

```rust
// Arrange things such that ASCII codepoints never
// result in the slower `is_xid_continue` getting called.
```